esphome/components/light/color_mode.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <cstdint>
+
+namespace esphome {
+namespace light {
+
+/// Color channels are the various outputs that a light has and that can be independently controlled by the user.
+enum class ColorChannel : uint8_t {
+  /// Intensity of a single white output.
+  WHITE = 1,
+  /// Color temperature of a single white output.
+  COLOR_TEMPERATURE = 2,
+  /// Intensities of a cold white and warm white output.
+  COLD_WARM_WHITE = 4,
+  /// Colored output using RGB format.
+  RGB = 8
+};
+
+// Define unary * operator to convert enum to underlying value -- this saves a lot of typing and ugly code.
+inline constexpr unsigned operator*(ColorChannel val) { return static_cast<uint8_t>(val); }
+
+/// Color modes are a combination of color channels that can be used at the same time.
+enum class ColorMode : uint8_t {
+  /// No color mode configured (cannot be a supported mode, is only active when light is either off
+  /// or does not support any color modes and channels).
+  UNKNOWN = 0,
+  /// White output.
+  WHITE = *ColorChannel::WHITE,
```

Comment:
```suggestion
  BRIGHTNESS = *ColorChannel::WHITE,
```

A light doesn't have to be white - for example consider a red LED (or warning light). It can be brightness controlled, but `WHITE` color mode wouldn't make sense.